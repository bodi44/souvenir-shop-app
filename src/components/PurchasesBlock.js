import React from 'react'

const PurchasesBlock = ({ date, allPurchases }) => {
  return (
    <div>
      <h5>{date}</h5>
      {allPurchases.map(purchase =>
        <p key={purchase.id}>
          {purchase.productName} {purchase.price} {purchase.currency}
        </p>)
      }
    </div>
  )
}

export default PurchasesBlock