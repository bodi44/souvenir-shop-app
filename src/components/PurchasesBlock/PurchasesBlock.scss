.PurchasesBlock {
  padding-left: 20px;
  margin-bottom: 10px;
  &__date {
    font-size: 18px;
    margin: 0 0 3px 0;
    padding: 0;
  }

  &__purchase {
    font-size: 16px;
    margin: 2px 0 0 0;
    padding: 0;
  }
}