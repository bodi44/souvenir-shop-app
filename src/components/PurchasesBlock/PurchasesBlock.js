import React from 'react'
import PropTypes from 'prop-types'

import './PurchasesBlock.scss'
import BEM from '../../helpers/BEM'

const b = BEM('PurchasesBlock')

const PurchasesBlock = ({ date, allPurchases }) => (
  <div className={b()}>
    <h5 className={b('date')}>{date}</h5>
    {allPurchases.map(purchase =>
      <p key={purchase.id} className={b('purchase')}>
        {purchase.productName} {purchase.price} {purchase.currency}
      </p>)
    }
  </div>
)

PurchasesBlock.propTypes = {
  date: PropTypes.string.isRequired,
  allPurchases: PropTypes.array.isRequired
}


export default PurchasesBlock