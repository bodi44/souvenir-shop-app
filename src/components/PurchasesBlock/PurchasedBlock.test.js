import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import expect from 'expect'
import Adapter from 'enzyme-adapter-react-16'

import PurchasesBlock from './PurchasesBlock'

Enzyme.configure({ adapter: new Adapter() })

describe('PurchaseBlock Component', () => {
  it('should render correctly with given date and allPurchases props', () => {
    const date = '2019-04-25'
    const allPurchases = [
      {
        id: 'c4f63a46-de46-49c3-9773-0a47c62cdb77',
        date: '2019-04-25',
        price: 12,
        currency: 'EUR',
        productName: 'Photo'
      }
    ]

    const component = shallow(<PurchasesBlock date={date} allPurchases={allPurchases}/>)

    expect(component).toMatchSnapshot()
  })
})