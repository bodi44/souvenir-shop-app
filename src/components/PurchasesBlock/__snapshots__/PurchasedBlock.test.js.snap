// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurchaseBlock Component should render correctly with given date and allPurchases props 1`] = `ShallowWrapper {}`;
